### PR TITLE
[WFCORE-5980] Deprecate / disable AbstractSubsystemBaseTest 'subsystem template' handling

### PR DESCRIPTION
**Pull request by lvydra**
https://issues.redhat.com/browse/WFCORE-5980

PR depends on https://github.com/wildfly/wildfly/pull/16252

### CODE REVIEW

File: subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java [hunk 1]
```diff
@@ -159,22 +160,17 @@ public void testSchema() throws Exception {
         SchemaValidator.validateXML(getSubsystemXml(), schemaPath, getResolvedProperties());
     }
 
+    @Test
+    public void testGetSubsystemTemplatePath() throws Exception {
+        Assume.assumeTrue(getSubsystemTemplatePaths().length == 0);
+    }
```
Comment (yersan) [inline subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java:166]: `Assume.assumeTrue` is only going to ignore the test if the condition is not satisfied. The Jira description suggests using this as throwaway work to detect which tests are invoking this method returning a non-empty String[].

You don't have to ignore the test if this condition does not occur. Just allow the testsuite to fail. Then create Jiras to clean up WildFly, and then, remove this `testGetSubsystemTemplatePath()` test and ignore both `testSchemaOfSubsystemTemplates()` and `getSubsystemTemplatePaths()`

That's my understanding of the task to accomplish here.

**Review by yersan — CHANGES_REQUESTED** [review]

File: subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java (hunk 1 above)
Comment (lvydra) [reply subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java]: Hi @yersan, thanks for clarification, updated.

**Comment by yersan** [comment]
Hold this one until WildFly full is ready for it. @lvydra ping me here when the work on full is done so we can move this on, thanks!

**Comment by yersan** [comment]
Hello @lvydra , notice Jobs results are cleaned as part of CI clean-up cycle. If you have not identified tests you need to modify on WildFly full, you will need to kick them off again to review them and continue with this Jira.

**Comment by lvydra** [comment]
Hi, @yersan, I have opened cleaning PR 

https://github.com/wildfly/wildfly/pull/16252

**Comment by lvydra** [comment]
Hi @yersan, https://github.com/wildfly/wildfly/pull/16252 is merged now.

File: subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java [hunk 2]
```diff
@@ -159,22 +160,17 @@ public void testSchema() throws Exception {
         SchemaValidator.validateXML(getSubsystemXml(), schemaPath, getResolvedProperties());
     }
 
+    @Test
+    public void testGetSubsystemTemplatePath() throws Exception {
+        Assert.assertTrue(getSubsystemTemplatePaths().length == 0);
+    }
+
```
Comment (yersan) [inline subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java:167]: This method can now be dropped.

File: subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java [hunk 3]
```diff
@@ -159,22 +160,17 @@ public void testSchema() throws Exception {
...
     /**
      * To enable a test for validation of the subsystem templates, override both {@link #getSubsystemXsdPath()} and {@link #getSubsystemTemplatePaths()}
      * then add a new test just calling this method.
      */
     protected void testSchemaOfSubsystemTemplates() throws Exception {
```
Comment (yersan) [inline subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java:172]: Add a @deprecate at `testSchemaOfSubsystemTemplates()` and in `getSubsystemTemplatePaths()`

**Review by yersan — CHANGES_REQUESTED** [review]
@lvydra CI is green, now you can continue with the pending changes

**Comment by lvydra** [comment]
Hi @yersan, thanks, updated.

File: subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java [hunk 4]
```diff
@@ -163,6 +164,7 @@ public void testSchema() throws Exception {
      * To enable a test for validation of the subsystem templates, override both {@link #getSubsystemXsdPath()} and {@link #getSubsystemTemplatePaths()}
      * then add a new test just calling this method.
      */
+    @Deprecated
     protected void testSchemaOfSubsystemTemplates() throws Exception {
```
Comment (yersan) [inline subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/AbstractSubsystemBaseTest.java:168]: @lvydra refers to the Jira description; testSchemaOfSubsystemTemplates turned into something that just throws AssumptionViolationException.

**Review by yersan — CHANGES_REQUESTED** [review]

**Comment by lvydra** [comment]
@yersan updated

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @lvydra !

**Pull request merged**